### DISCUSSION TITLE
Can't make `$XDG_RUNTIME_DIR/app` readable

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [netblue30/firejail#5150] [Q&A] [answered]
I have this in `firefox.local`:

```
# For KeePassXC:
noblacklist ${RUNUSER}/app/
noblacklist ${RUNUSER}/app/org.keepassxc.KeePassXC/
mkdir       ${RUNUSER}/app/org.keepassxc.KeePassXC/
whitelist   ${RUNUSER}/app/
whitelist   ${RUNUSER}/app/org.keepassxc.KeePassXC/
```

But inside the sandbox, `app` is not user-readable:

```
$ ls -l $XDG_RUNTIME_DIR/
total 0
dr-------- 2 nobody nobody 40 May 17 18:31 app
-r-------- 1 nobody nobody  0 May 17 18:31 bus
drwx------ 2 hugo   hugo   60 May 17 18:31 dconf
srw-rw-rw- 1 hugo   hugo    0 May 17 18:28 pipewire-0
drwxr-xr-x 2 hugo   hugo   60 May 17 19:16 pulse
srwxr-xr-x 1 hugo   hugo    0 May 17 18:28 wayland-1
```

Any ideas what's wrong? The directory belongs to `hugo:hugo` outside the sandbox.

**Comment by rusty-snake** [comment]
`writable-run-user`

**Reply by WhyNotHugo** [reply]
Huh? From `firejail-profile`:

```
       writable-run-user
              Disable   the   default   blacklisting   of    run/user/$UID/systemd    and
              /run/user/$UID/gnupg.
```

Is that out of date?

**Reply by WhyNotHugo** [reply]
Doesn't look like that `writable-run-user` covers this directory too: https://github.com/netblue30/firejail/blob/baeec77b760cb7789dba99f885d0d65a817a335b/src/firejail/fs.c#L789-L814

**Reply by rusty-snake** [reply]
Wait why is this then named `writeable`-`run-user`?!

**Comment by rusty-snake** [comment]
Anyway this is the blacklist you need to `noblacklist`:

https://github.com/netblue30/firejail/blob/78aaed11d85be690204fbd956732a7c1f18363e3/etc/inc/disable-common.inc#L595

**Comment by WhyNotHugo** [comment, marked as answer]
Oh, it **is** `noblacklist`d, but the trailing slash is significant, so `noblacklist ${RUNUSER}/app/` is not the same as `noblacklist ${RUNUSER}/app`. Removing the trailing slash fixed it.
